HeterogeneousCore/CUDAUtilities/interface/maxCoopBlocks.h

```diff
@@ -0,0 +1,30 @@
+
+#ifndef HeterogeneousCore_CUDAUtilities_interface_maxCoopBlocks_h
+#define HeterogeneousCore_CUDAUtilities_interface_maxCoopBlocks_h
+
+#include <cuda_runtime.h>
+#include <algorithm>
+#include <cstdlib>
+
+template <typename F>
+inline int maxCoopBlocks(F kernel, int nthreads, int shmem, int device, int redFact = 10) {
+#define GET_COOP_RED_FACT_FROM_ENV
+
+// to drive performance assessment by envvar
+#ifdef GET_COOP_RED_FACT_FROM_ENV
+  auto env = getenv("COOP_RED_FACT");
```

Comment:
We recently had a campaign for
```suggestion
  auto env = std::getenv("COOP_RED_FACT");
```
to make sure these calls are thead safe.